app/src/main/java/org/p2p/wallet/striga/onboarding/StrigaOnboardingFragment.kt

```diff
@@ -0,0 +1,187 @@
+import org.p2p.uikit.utils.text.bind
+import org.p2p.wallet.R
+import org.p2p.wallet.auth.repository.Country
+import org.p2p.wallet.common.mvp.BaseMvpFragment
+import org.p2p.wallet.databinding.FragmentStrigaOnboardingBinding
+import org.p2p.wallet.intercom.IntercomService
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.viewbinding.getColor
+import org.p2p.wallet.utils.viewbinding.getDrawable
+import org.p2p.wallet.utils.viewbinding.getString
+import org.p2p.wallet.utils.viewbinding.viewBinding
+
+class StrigaOnboardingFragment :
+    BaseMvpFragment<StrigaOnboardingContract.View, StrigaOnboardingContract.Presenter>(
+        R.layout.fragment_striga_onboarding
+    ),
+    StrigaOnboardingContract.View {
+
+    companion object {
+        fun create(): StrigaOnboardingFragment = StrigaOnboardingFragment()
+    }
+
+    override val presenter: StrigaOnboardingContract.Presenter by inject()
+
+    private val binding: FragmentStrigaOnboardingBinding by viewBinding()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        handleBackPress()
+        bindHelpText()
+
+        binding.buttonContinue.setOnClickListener {
+            presenter.onClickChangeCountry()
+        }
+        binding.textViewPoweredBy.setOnClickListener {
+            Intent(Intent.ACTION_VIEW, getString(R.string.striga_powered_by_url).toUri())
+                .also { startActivity(it) }
+        }
+    }
+
+    override fun navigateNext() {
+        // TODO: navigate to next screen
+    }
+
+    override fun setCurrentCountry(country: Country) {
+        mapCountryView(country.name, country.flagEmoji)
+    }
+
+    override fun setAvailabilityState(state: StrigaOnboardingContract.View.AvailabilityState) {
+        when (state) {
+            StrigaOnboardingContract.View.AvailabilityState.Available -> handleAvailableState(state)
+            StrigaOnboardingContract.View.AvailabilityState.Unavailable -> handleUnavailableState(state)
+        }
+
+        binding.image.animate()
+            .alpha(1f)
+            .setDuration(150L)
+            .start()
+    }
+
+    override fun openCountrySelection() {
+        /* TODO: select country
+            replaceFragmentForResult(Fragment(), "select_country", onResult = { requestKey, bundle ->
+                val country = bundle.getParcelableCompat<Country>("country")
+                presenter.onCountrySelected(country)
+            })
+        */
+    }
+
+    override fun openHelp() {
+        IntercomService.showMessenger()
+    }
+
+    private fun handleBackPress() {
+        binding.toolbar.setNavigationOnClickListener { onBackPressed() }
+        requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner) {
+            onBackPressed()
+        }
+    }
+
+    private fun onBackPressed() {
+        popBackStack()
+    }
+
+    private fun bindHelpText() {
```

Comment:
can be extension method of binding or wrapped in with(binding)

Reply:
I'll wrap it in `with`